src/synthesizer/imaging/extensions/circular_aperture.c

```diff
@@ -0,0 +1,422 @@
+/******************************************************************************
+ * C extension for calculating the overlap area between a circle and a pixel.
+ *
+ * This is heavily adapted from the circular_aperture module in the photutils
+ * package. The original code is available at:
+ * github.com/astropy/photutils/tree/ee0dd3ee7b4bc0659f38262b558ca0d6d6836bf5
+ * The original code is licensed under a BSD 3-Clause License, which is
+ * included in the LICENSE file in the linked repository.
+ *****************************************************************************/
+
+/* C includes */
+#include <math.h>
+#include <stdlib.h>
+
+#ifdef WITH_OPENMP
+#include <omp.h>
+#endif
+
+/* Python includes */
+#include <Python.h>
+#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+#include <numpy/arrayobject.h>
+
+/* Local includes */
+#include "../../extensions/property_funcs.h"
+#include "../../extensions/timers.h"
+
+/* It's possible we don't have PI... if so define it. */
+#ifndef M_PI
+#define M_PI 3.141592653589793238462643383279502884
```

Comment:
Do we need that much precision?

Reply:
Why not ;). To be fair, this will rarely be used since it'll be defined most of the time.